src/Microsoft.AspNet.Mvc.ModelBinding/BinderMetadata/IBinderTypeProviderMetadata.cs

```diff
@@ -9,12 +9,12 @@ namespace Microsoft.AspNet.Mvc.ModelBinding
     /// Provides a <see cref="Type"/> which implements <see cref="IModelBinder"/> or 
     /// <see cref="IModelBinderProvider"/>.
     /// </summary>
-    public interface IBinderTypeProviderMetadata : IBinderMetadata
+    public interface IBinderTypeProviderMetadata : IBindingSourceMetadata
     {
         /// <summary>
         /// A <see cref="Type"/> which implements either <see cref="IModelBinder"/> or
         /// <see cref="IModelBinderProvider"/>.
         /// </summary>
-        Type BinderType { get; set; }
+        Type BinderType { get; }
```

Comment:
Interfaces like this shouldn't be get/set. The attribute still allows you to set it.